src/Components/HomePage.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import getCountries from '../redux/api';
+import Country from './Country';
+import Header from './HomePageHeader';
+
+const HomePage = () => {
+  const dispatch = useDispatch();
+  const countries = useSelector((state) => state.countries.countries);
+
+  const [searchItem, setSearchItem] = useState('');
+
+  useEffect(() => {
+    if (countries.length === 0) {
+      dispatch(getCountries());
+    }
+  }, [dispatch, countries.length]);
+
+  const newCountries = countries.filter((country) => (
+    country.name.toLowerCase().includes(searchItem.toLowerCase())
+    || country.region.toLowerCase().includes(searchItem.toLowerCase())));
+
+  const handleSearch = (e) => {
+    e.preventDefault();
+    setSearchItem(e.target.value);
+  };
+
+  return (
+    <>
+      <Header />
+      <main>
+        <input
+          className="search-bar"
+          type="text"
+          name="searchItem"
+          placeholder="search for a country"
+          value={searchItem}
+          onChange={handleSearch}
+        />
+        <div className="card-container">
+          {
+            searchItem.length
+              ? newCountries.map((country) => (
+                <Country
+                  key={uuidv4()}
+                  data={country}
+                />
+              ))
+              : countries.map((country) => (
+                <Country
+                  key={uuidv4()}
+                  data={country}
+                />
+              ))
+          }
+        </div>
+      </main>
+    </>
+  );
```

Comment:
- [ ] it would be better to stick to the given design as required and try to remove the spaces between cards as in the design it should be no spaces between them in order to match the design 
- Screenshot of the requirements 
<img width="529" alt="Screenshot 2023-02-14 at 15 43 22" src="https://user-images.githubusercontent.com/61790587/218771023-dea2e0ee-43b3-4e36-b3d9-809154d73581.png">


- Yours 
<img width="320" alt="Screenshot 2023-02-14 at 15 34 34" src="https://user-images.githubusercontent.com/61790587/218771214-6219626f-e746-4467-b574-3bd806aa6621.png">


- Design 
<img width="196" alt="Screenshot 2023-02-14 at 15 39 06" src="https://user-images.githubusercontent.com/61790587/218771160-e9d5d0b8-c213-49b9-a820-f1defdd548b7.png">

Reply:
The project has been updated to meet the given style guide. Thanks for your feedback😇

Reply:
nice job :+1: